R/get_recommended_interventions_higher_order_cost.R

```diff
@@ -1,99 +1,90 @@
+#' @param beta_vec numeric vector: coefficient estimates for the intervention
+#' components, including the intercept term, but not any coefficient
 #' estimates for the center characteristics.
 #' For example: c(0.1, 0.3, 0.15).
-#' If intercept == T, then the first element 0.1 is the estimate for the intercept.
-#' @param cost_list_of_lists A nested list structure defining cost functions for
+#' @param center_cha_coeff_vec numeric vector: coefficients estimates for the
+#' center characteristics.
+#' @param center_cha numeric vector: given values of the center characteristics
+#' that will be used in LAGO optimizations.
+#' @param cost_list_of_lists list: a nested list structure defining cost functions for
 #' the intervention components. Each sublist represents a component and contains
 #' coefficients for its cost function.
-#' @param intervention_lower_bounds numeric vector, Lower bounds for the
-#' intervention components.
-#' For example: c(0,0).
-#' @param intervention_upper_bounds numeric vector, Upper bounds for the
-#' intervention components.
-#' For example: c(10,20).
-#' @param outcome_goal numeric, The outcome goal.
-#' @param outcome_goal_optimization character, Method used behind the scenes for
-#' calculating the recommended interventions. Either "numerical" or "grid_search".
-#' @param grid_search_step_size numeric, step size for the grid search algorithm,
-#' default value is 0.01 for each intervention component.
-#' @param center_cha_coeff_vec numeric vector, Coefficients estimates for the
-#' center characteristics.
-#' For example: c(0.2, 0.4).
-#' @param center_cha numeric vector, Given values of the center characteristics.
-#' @param intercept boolean, Whether the intercept was included in the fitted model.
-#' @param phase character, Either "Planning" or "Optimal". "Planning" means
-#' we are calculating the recommended interventions for the next stage. When the
-#' outcome goal cannot be reached, we aim to bring the estimated outcome mean/probability
-#' for the intervention group in the next stage to be as close to the outcome goal
-#' as possible. "Optimal" means we are calculating the optimal interventions at the
-#' end of the stage, using the idea from web appendix section 5.1 of Nevo et al.
-#'
-#' For now, this defaults to "planning". we can discuss whether we need
-#' both "Planning" and "Optimal".
+#' @param intervention_lower_bounds numeric vector: lower bounds of the intervention components.
+#' For example: for a two-component intervention package, lower bounds could be
+#' c(0,0).
+#' @param intervention_upper_bounds numeric vector: upper bounds of the intervention components.
+#' For example: for a two-component intervention package, upper bounds could be
+#' c(10,20).
+#' @param outcome_goal numeric: the outcome goal, a desired probability or mean
+#' value.
+#' @param outcome_goal_opt_method character: the method used for calculating the
```

Comment:
I assume all of these parameters are required? Is there a default for grid search or numerical? Maybe we can add a very short description (1-2 sentences) of when to use "numerical" or "grid_search" and vice versa? I think some folks from other fields might not be familiar with "grid_search" or can conceptualize it to relate to their data. For example,
@param search_method A `character` string. Specifies the method used for optimizing the model or parameters. Must be either `"grid_search"` or `"numerical"`. Default is `"numerical"`.
#'   - Use `"grid_search"` if you want to evaluate the model over a predefined set of parameter combinations. This method is useful when you have specific ranges or values to explore systematically.
#'   - Use `"numerical"` for optimization methods that rely on gradient-based or other iterative techniques to find the best parameters without exhaustively searching a predefined grid. This is generally faster and suitable for larger datasets or when the parameter space is too large for grid search.
or even shorter than this. We can make it even more relatable in the clinical trial context.

Reply:
agreed, it should be more clear. The default should be numerical which is passed down from the upper level. `R/get_recommended_interventions_higher_order_cost.R` is not a user facing function, but i think you brought up a good point, i will add the clarifications to the corresponding parameter in the user facing function `R/calculate_recommended_interventions.R`.